runtime/lua/vim/lsp/util.lua

```diff
@@ -1154,34 +1163,29 @@ function M.fancy_floating_markdown(contents, opts)
   api.nvim_win_set_option(winnr, 'conceallevel', 2)
   api.nvim_win_set_option(winnr, 'concealcursor', 'n')
 
+  vim.cmd("ownsyntax lsp_markdown")
+
   local idx = 1
   --@private
   local function apply_syntax_to_region(ft, start, finish)
     if ft == "" then
-      vim.cmd(string.format("syntax region markdownCodeBlock start=+\\%%%dl+ end=+\\%%%dl+ keepend extend", start, finish + 1))
+      vim.cmd(string.format("syntax region markdownCode start=+\\%%%dl+ end=+\\%%%dl+ keepend extend", start, finish + 1))
       return
     end
     local name = ft..idx
     idx = idx + 1
     local lang = "@"..ft:upper()
+    -- HACK: reset current_syntax, since some syntax files like markdown won't load if it is already set
```

Comment:
I'm not sure this is a good idea, this callback is asynchronous so  I'm not sure we're guaranteed that we'll delete `current_syntax` on the floating win

Reply:
right, should be `bufnr` instead

Reply:
One alternative (given without comment0 is we could have loading lsp_markdown.vim clear the variable *then* load markdown to avoid doing it here

Reply:
ah no, the issue is not just with `markdown`. Most syntax files wont load if there's already a `current_syntax` set.

See the javacript syntax for exanmple:   https://github.com/neovim/neovim/blob/75992f2ad9189a5f39b3e2ae7962c68d3af815a4/runtime/syntax/javascript.vim#L19

So loading any other syntax file to apply to a region, might skip loading that syntax altogether